app/controllers/discourse_ai/sentiment/sentiment_controller.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module DiscourseAi
+  module Sentiment
+    class SentimentController < ::Admin::StaffController
+      include Constants
+      requires_plugin ::DiscourseAi::PLUGIN_NAME
+      requires_login
+
+      def posts
+        group_by = params[:group_by]&.to_sym
+        group_value = params[:group_value].presence
```

Comment:
because group_by and group_value are required maybe you can rely on strong parameters
```
 group_by = params.required(:group_by)&.to_sym
```